chapter_frame_test.go

```diff
@@ -29,129 +29,159 @@ func prepareTestFile() (*os.File, error) {
+				},
+			},
+			wantElementId:   "chap0",
+			wantTitle:       "chapter 0",
+			wantDescription: "",
+		},
+		{
+			name: "with description",
+			fields: fields{
+				ElementID:   "chap0",
+				StartTime:   0,
+				EndTime:     time.Duration(1000 * nanosInMillis),
+				StartOffset: 0,
+				EndOffset:   0,
+				Description: &TextFrame{
+					Encoding: EncodingUTF8,
+					Text:     "chapter 0",
+				},
+			},
+			wantElementId:   "chap0",
+			wantTitle:       "",
+			wantDescription: "chapter 0",
+		},
+		{
+			name: "with title and description",
+			fields: fields{
+				ElementID:   "chap0",
+				StartTime:   0,
+				EndTime:     time.Duration(1000 * nanosInMillis),
+				StartOffset: 0,
+				EndOffset:   0,
+				Title: &TextFrame{
+					Encoding: EncodingUTF8,
+					Text:     "chapter 0 title",
+				},
+				Description: &TextFrame{
+					Encoding: EncodingUTF8,
+					Text:     "chapter 0 description",
+				},
+			},
+			wantElementId:   "chap0",
+			wantTitle:       "chapter 0 title",
+			wantDescription: "chapter 0 description",
+		},
+		{
+			name: "non-zero time and offset",
+			fields: fields{
+				ElementID:   "chap0",
+				StartTime:   time.Duration(1000 * nanosInMillis),
+				EndTime:     time.Duration(1000 * nanosInMillis),
+				StartOffset: 10,
+				EndOffset:   10,
+			},
+			wantElementId:   "chap0",
+			wantTitle:       "",
+			wantDescription: "",
 		},
 	}
-	tag.AddChapterFrame(chap)
-
-	if err := tag.Save(); err != nil {
-		t.Error(err)
-	}
-	tag.Close()
-
-	tag, err = Open(tmpFile.Name(), Options{Parse: true})
-	if tag == nil || err != nil {
-		log.Fatal("Error while opening mp3 file: ", err)
-	}
-	frame := tag.GetLastFrame("CHAP").(ChapterFrame)
-	if frame.ElementID != "chap0" {
-		t.Error(err)
-	}
-	if frame.SubTitle.Text != "chapter 0" {
-		t.Errorf("expected: %s, but got %s", "chapter 0", frame.Title)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tmpFile, err := prepareTestFile()
+			if err != nil {
+				t.Error(err)
+			}
+			defer os.Remove(tmpFile.Name())
+
+			tag, err := Open(tmpFile.Name(), Options{Parse: true})
+			if tag == nil || err != nil {
+				log.Fatal("Error while opening mp3 file: ", err)
+			}
+
+			cf := ChapterFrame{
+				ElementID:   tt.fields.ElementID,
+				StartTime:   tt.fields.StartTime,
+				EndTime:     tt.fields.EndTime,
+				StartOffset: tt.fields.StartOffset,
+				EndOffset:   tt.fields.EndOffset,
+				Title:       tt.fields.Title,
+				Description: tt.fields.Description,
+			}
+			tag.AddChapterFrame(cf)
+
+			if err := tag.Save(); err != nil {
+				t.Error(err)
+			}
+			tag.Close()
+
+			tag, err = Open(tmpFile.Name(), Options{Parse: true})
+			if tag == nil || err != nil {
+				log.Fatal("Error while opening mp3 file: ", err)
+			}
+			frame := tag.GetLastFrame("CHAP").(ChapterFrame)
+			if frame.ElementID != tt.wantElementId {
+				t.Errorf("expected: %s, but got %s", tt.wantElementId, frame.ElementID)
+			}
+			if frame.Title.Text != tt.wantTitle {
```

Comment:
Why not reusing `tt.fields.Title.Text` instead of `tt.wantTitle`?
```go
if tt.fields.Title != nil && frame.Title.Text != tt.fields.Title.Text {
   ...
}
```